### PR TITLE
[docs]: Create Deprecation page per deprecation group

### PR DESCRIPTION
**Pull request by niklas-wortmann**
This is a rough POC to see if it works for our purpose to improve the deprecation messages.

Text might need some improvement/additional information
@benlesh @cartant

### CODE REVIEW

File: src/internal/observable/combineLatest.ts [hunk 1]
```diff
@@ -44,19 +44,19 @@ export function combineLatest<O1 extends ObservableInput<any>, O2 extends Observ
 export function combineLatest<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>, O5 extends ObservableInput<any>, O6 extends ObservableInput<any>, R>(v1: O1, v2: O2, v3: O3, v4: O4, v5: O5, v6: O6, resultSelector: (v1: ObservedValueOf<O1>, v2: ObservedValueOf<O2>, v3: ObservedValueOf<O3>, v4: ObservedValueOf<O4>, v5: ObservedValueOf<O5>, v6: ObservedValueOf<O6>) => R, scheduler?: SchedulerLike): Observable<R>;
 
 // With a scheduler (deprecated)
-/** @deprecated Passing a scheduler here is deprecated, use {@link subscribeOn} and/or {@link observeOn} instead */
+/** @deprecated The scheduler argument is deprecated, use {@link subscribeOn} and/or {@link observeOn}. Details: {@link https://rxjs.dev/deprecations/scheduler-argument} */
```
Comment (cartant) [inline src/internal/observable/combineLatest.ts:47]: Is the content of the `deprecated` tag shown anywhere in the docs? AFAICT, it isn't. If that is the case, can we please remove this `{@link ... }` business, as it just makes the messages harder to read in linter reports and in the VS Code hints.

Comment (niklas-wortmann) [reply src/internal/observable/combineLatest.ts]: I changed it for now, but in the long run I would like to have this in the docs as well. 
In the long run I would like to have this in the docs and IntelliJ is supporting @link and @see, so it's mainly just vscode that has several open issues for that:
https://github.com/microsoft/TypeScript/issues/35524
https://github.com/microsoft/TypeScript/issues/16498

As soon as this is supported by VsCode I'd like to change it to @link etc again

File: docs_app/content/guide/higher-order-observables.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+```ts
+const fileObservable = urlObservable.pipe(
+   map(url => http.get(url)),
+);
+```
+
+`http.get()` returns an Observable for each URL. Now you have an Observable *of* Observables, a higher-order Observable.
+
+But how do you work with a higher-order Observable? Typically, by _flattening_: by converting a higher-order Obser vable into an ordinary Observable. For example:
```
Comment (jakovljevic-mladen) [inline docs_app/content/guide/higher-order-observables.md:13]: Hi. There's an extra space: `Obser vable` -> `Observable`

File: docs_app/content/guide/higher-order-observables.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+```ts
+const fileObservable = urlObservable.pipe(
+   concatMap(url => http.get(url)),
+);
+```
+
+The Observable returned in the `concatMap` function is usually referred to as a so-called "inner" Observable, while in this context the `urlObservable` is the so-called "outer" Observable.
+
+The [`concatMap()`](/api/operators/concatMap) operator subscribes to each "inner" Observable, buffers all further emissions of the "outer" Observable, and copies all the emitted values until the inner Observable completes, and continues processing the values of the "outer Observable". All of the values are in that way concatenated.  Other useful flattening operators are
```
Comment (jakovljevic-mladen) [inline docs_app/content/guide/higher-order-observables.md:23]: I'm sorry for bothering as this is probably not important since browsers clear more than one space, but there are two spaces between `concatenated.` and `Other` words.

File: docs_app/content/guide/higher-order-observables.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+* [`mergeMap()`](/api/operators/mergeMap) — subscribes to each inner Observable as it arrives, then emits each value as it arrives
```
Comment (jakovljevic-mladen) [inline docs_app/content/guide/higher-order-observables.md:25]: This bullet is missing the dot at the end of sentence (unlike other two below which are not). Or maybe, since these bullets are continuation of the sentence `Other useful flattening operators are`, a comma `,` or semicolon `;` could be used for `switchMap` and `switchMap`.

File: docs_app/content/deprecations/scheduler-argument.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+// Deprecated approach
+of([1,2,3], asyncScheduler).subscribe(x => console.log(x));
+// suggested approach
+scheduled(of([1,2,3]), asyncScheduler).subscribe(x => console.log(x));
+```
+
+### Refactoring of `merge`, `concat`, `startWith` and `endWith`
+
+In case you used to pass a scheduler argument to one of these operators you propably had code like this:
```
Comment (jakovljevic-mladen) [inline docs_app/content/deprecations/scheduler-argument.md:47]: `propably` -> `probably`.

**Comment by cartant** [comment]
Fixed some typos that were pointed out above and rebased onto master.

There are some deprecation messages that still include the `{@ ... }` markup that's discussed [here](https://github.com/ReactiveX/rxjs/pull/5426#discussion_r425513689). IMO, it would be best to remove these ATM, as they clutter the output in VS code and on the command line. Apart from that, this LGTM and, IMO, improvements regarding the deprecations need to be incorporated into the current v6 docs ASAP.

**Comment by cartant** [comment]
FWIW, we really should try to get the deprecation messages sorted before this lands:

https://mobile.twitter.com/WenluWang/status/1275458404132073473

We should really fix them in v6 too, as people are going to be using that for a while in Angular, etc.

File: src/internal/observable/from.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { ObservableInput, SchedulerLike, ObservedValueOf } from '../types';
 import { scheduled } from '../scheduled/scheduled';
 
 export function from<O extends ObservableInput<any>>(input: O): Observable<ObservedValueOf<O>>;
-/** @deprecated use {@link scheduled} instead. */
+/** @deprecated The scheduler argument is deprecated, use scheduled. Details: https://rxjs.dev/deprecations/scheduler-argument */
```
Comment (benlesh) [inline src/internal/observable/from.ts:7]: Do backticks not work here?

Comment (benlesh) [reply src/internal/observable/from.ts:7]: I think my concern is this reads like we're saying "Don't use a scheduler, use scheduled" and to people who don't yet know about `scheduled` it might read like "don't schedule, schedule!" haha.

Comment (niklas-wortmann) [reply src/internal/observable/from.ts:7]: backticks and general md support is IDE specific, we could just rephrase it to 'use the scheduled operator' and I think this way it's pretty clear

File: docs_app/content/deprecations/scheduler-argument.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+```ts
+import { combineLatest, timer, asyncScheduler } from 'rxjs'; 
+import { subscribeOn } from 'rxjs/operators'
+
+combineLatest(
+  [timer(0, 1000), timer(500, 500)],
+).pipe(
+  subscribeOn(asyncScheduler)
+).subscribe(console.log)
+```
```
Comment (cartant) [inline docs_app/content/deprecations/scheduler-argument.md:101]: Just realised that there's a `combineAll` operator that's the higher-order version of `combineLatest`, so the equivalent ought to be:

```ts
import { scheduled, of, asyncScheduler } from 'rxjs'; 
import { combineAll } from 'rxjs/operators'
scheduled(
  [timer(0, 1000), timer(500, 500)],
  asyncScheduler
).pipe(
  combineAll()
).subscribe(x => console.log(x));
```

And maybe this example should use the `[of('hello '), of('World')]` sources that were used above? Was there a particular reason for using the timers?

**Review by cartant — APPROVED** [review]
